NOTICE

```diff
@@ -0,0 +1,4 @@
+Thanos
+Copyright (c) 2022 The Thanos Authors
+
+This project includes software developed at Oracle America, Inc.
```

Comment:
https://infra.apache.org/licensing-howto.html

> Do not modify LICENSE or NOTICE for non-bundled dependencies.

I rather see this go. If every person or company starts doing it, it becomes a mess.

Reply:
Hi @wiardvanrij 
The guideline is from Oracle's legal counsel who is following the CFCF and Apache guideline.

CNCF doc: https://github.com/cncf/foundation/blob/main/copyright-notices.md 
Apache License  doc: https://infra.apache.org/licensing-howto.html

I will check with our legal team and see if there is anything that we can do on this.

Thx!

Reply:
Hey folks, I'm one of the open source advocates at Oracle and we are working with legal on this (and other) copyright-related requirements, but for now, this is a mandated requirement for our contributions to CNCF projects.

Reply:
_With my maintainer hat on;_ 

Sorry if I'm being hesitant but my priority is the project, community, and users. Perhaps it's not a big deal and this done soon.  As I said, I've pinged others as well to see what we have to do :) 

_With my personal hat on;_

Why? As noted in those guidelines you provide, it causes a burden on OSS projects/maintainers. Even your own employee is blocked for weeks/months with legal. It's IMO clear that you retain the copyright. 

That said, I honestly admire the effort here to contribute to the project. It's awesome to have quality code improvements from companies. Maybe we as a project have to do more to enable the "we got a legal department" companies in order to be more easily involved in open source without constraints. @Djelibeybi , I would personally appreciate the reasoning for the mandate. What is the legal thing that has to be covered here and what can we do as a project to make that easier? 

Anyhow, still thanks a lot @aarontams for your work (:  We will get this done!

Reply:
@wiardvanrij With (mostly) my personal hat on: there is no clear record of who "The Thanos Authors" _are_ outside of git that can be pointed at with zero ambiguity as to the provenance of the code _when it was contributed_ so when you say "it's .. clear that you retain the copyright" it's _not_ clear who "you" is in this case: it assumes that Git and GitHub are available for cross-reference.

But if this source code is distributed without any Git metadata, which happens a lot when packaged for distros, or when distributed as a container imagte, there is no way to prove the retention of copyright beyond "The Thanos Authors" and you're not distributing a handy list of those authors.

Reply:
Hi All, 

Unfortunately, such license entry is a precedence for us and might have unforeseen implications for our users. We are not lawyers, so we can't approve this. 

We added a ticket to CNCF to check if adding this line is possible, sorry for the inconvenience.

Reply:
Hi all.  Just sticking my nose in here, so I can watch this one issue.  The CNCF's response here is particularly relevant to me.